### PR TITLE
Add reflection fit and more generalizations

### PR DESCRIPTION
**Pull request by lneuhaus**
Adds a reflection signal to FitFabryPerot.

### CODE REVIEW

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 1]
```diff
@@ -104,18 +83,19 @@ def calibrate(self, autosave=False):
         else:
             return None
 
-    def fit_with_voigt(self, curve, voltage):
+    def fit(self, curve, voltage):
```
Comment (heltluke) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:86]: Why change the name? Perhaps other fits, e.g. including thermal effects, will be necessary.

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 2]
```diff
@@ -104,18 +83,19 @@ def calibrate(self, autosave=False):
...
-        p_params = peak.guess(curve, x=x_for_fit)
+        p_params = peak.guess(curve, x=x_for_fit, negative=self.negative)
         background = ConstantModel()
         b_params = background.guess(curve)
         background.set_param_hint("c", value=np.min(curve))
         mod = peak + background
         pars = p_params + b_params
-        result = mod.fit(curve, pars, x=x_for_fit, method="least_squares")
+        result = mod.fit(curve, pars, x=x_for_fit, method="least_squares", 
+                         max_nfev=self.max_nfev)
```
Comment (heltluke) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:98]: Nice. I think it's very useful to set a limit here. Somewhat bizarre that the default is `None`.

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 3]
```diff
@@ -104,18 +83,19 @@ def calibrate(self, autosave=False):
...
             return None
 
-    def fit_with_voigt(self, curve, voltage):
+    def fit(self, curve, voltage):
         """ Fits curve to a Voigt profile. """
         x_for_fit = self.voltage_to_fit(voltage)
 
         peak = VoigtModel()
-        p_params = peak.guess(curve, x=x_for_fit)
+        p_params = peak.guess(curve, x=x_for_fit, negative=self.negative)
```
Comment (heltluke) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:91]: Cool! I had no idea this was there. I see it in the [lmfit code](https://github.com/lmfit/lmfit-py/blob/82fd7742d184bc4f31b0604740f48d1b265c3c8b/lmfit/models.py#L450), but (at least not easily) in the lmfit docs. Here's hoping we don't need to touch  the maxima and minima. :crossed_fingers:

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 4]
```diff
@@ -280,14 +285,42 @@ def _bandwidth_in_m(self):
...
 class FitFabryPerot(FabryPerot):
     inputs = LockboxModuleDictProperty(transmission=FitFPTransmission,
+                                       reflection=FitFPReflection,
                                        pdh=FPPdh)
 
-class FitVsqFabryPerot(FabryPerot):
-    inputs = LockboxModuleDictProperty(transmission=FitFPTransmissionVsq,
-                                       pdh=FPPdh)
 
+class NonlinearFitInput(FitInput):
```
Comment (heltluke) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:302]: Nice generalization!

File: pyrpl/software_modules/lockbox/models/fabryperot.py [hunk 5]
```diff
@@ -89,6 +65,9 @@ def calibrate(self, autosave=False):
         self.calibration_data.max = maximum
         self.calibration_data.mean = curve.mean()
         self.calibration_data.rms = curve.std()
+        self.calibration_data.center = centre_voltage
```
Comment (heltluke) [inline pyrpl/software_modules/lockbox/models/fabryperot.py:68]: Should maybe fix AE vs BE (likely my mistake for introducing BE).

File: pyrpl/software_modules/lockbox/models/fabryperot.py (hunk 1 above)
Comment (lneuhaus) [reply pyrpl/software_modules/lockbox/models/fabryperot.py:86]: I renamed it to make this implementation an example of a general implementation of an input with a fit, where the details hidden behind the "fit" method would encode the type of fit, to be reflected in the class name later on. So if we had a second fit type, I would rather go with a class `VoigtFitFPInput` with a method `fit`, than a `FitFPInput` class with method `fit_with_voigt`.

Comment (lneuhaus) [reply pyrpl/software_modules/lockbox/models/fabryperot.py:86]: and with the way its written here, we can simply inherit from any Fit Input class, and replace the fit method to have a different fit implemented.

File: pyrpl/software_modules/lockbox/models/fabryperot.py (hunk 3 above)
Comment (lneuhaus) [reply pyrpl/software_modules/lockbox/models/fabryperot.py:91]: up, you're right..

**Pull request merged**